server/src/modules/cats/cats.controller.ts

```diff
@@ -1,23 +1,37 @@
 import { Controller, Get, Post, Body, Param, Query } from '@nestjs/common';
 import { CatsService } from './cats.service';
-import { Cat } from 'src/models/cat/cat.model';
+import type { Cat } from 'src/models/cat/cat.model';
+import { CreateCatDto } from 'src/types/dto/cat.dto';
 
 @Controller('cats')
 export class CatsController {
     constructor(private readonly catService: CatsService) { }
 
     @Get()
     async findAll(@Query('name') name?: string): Promise<Cat[]> {
-        return this.catService.findAll(name);
+        try {
+            return this.catService.findAll(name);
+        } catch (error) {
+            console.log(error);
+        }
```

Comment:
instead of adding `try catch` to all functions you can use:
https://docs.nestjs.com/exception-filters